Objects/abstract.c

```diff
@@ -755,9 +755,9 @@ PyObject_Format(PyObject *obj, PyObject *format_spec)
 int
 PyNumber_Check(PyObject *o)
 {
-    return o && o->ob_type->tp_as_number &&
-           (o->ob_type->tp_as_number->nb_int ||
-        o->ob_type->tp_as_number->nb_float);
+    return o && Py_TYPE(o)->tp_as_number &&
```

Comment:
Maybe it would be worth to help the compiler to store Py_TYPE() into a variable here, instead of "reading" the type 3 times?

(On this specific example, be carefull since o can be NULL ;-))